src/content/app/tools/blast/components/blast-settings/BlastSettings.tsx

```diff
@@ -287,13 +290,26 @@ const buildSelect = (setting: {
   options: { label: string; value: string }[];
   label: string;
   selectedOption: string;
+  limit?: ParameterRange;
+  blastProgram?: BlastProgram;
```

Comment:
The `buildSelect` doesn't need to know all these. Instead, we should have a dedicated function that transforms the combination of the blast program and the max/min values into an array of options, of the shape defined on line 290.